### PR TITLE
Adding comment to each string ressources (Improving translation process)

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Currently, AnkiDroid is quite hard to translate. Because most strings have little to no context. As an example "Empty deck" may appears on a button to "empty this deck", or on a warning to state that "this deck is empty". And there is no way for translators to know which one is meant. Actually, I can imagine that they guess it's the first one and don't realize it may mean the second one. 

A usual translator will probably have little idea of what means "apkg failed validation" unless they know internals very well. 

Actually, there are some strings I don't even understand myself, even with codebase access, knowing ankidroid quite well I believe, and knowing an important part of the codebase.

My point is that almost all strings should get a comment for translators. Personally, either the string is easy to comment and I can do it taking little time. Or I don't understand the string, I need to go in the code to see where it's used, and this prove that the comment was actually useful.

Furthermore, if each string has a comment, this will show the correct way to behaves to new contributors and they'll directly add comments with their strings, so this work won't have to be repeated.

This is my goal here. Not yet done, but showing what was done over the week-end will allow for feedback. It's based over https://github.com/ankidroid/Anki-Android/pull/6982 . Without this other PR, it would be a nightmare to do any conflict resolution.

One commit delete ressources that are used nowhere. We can remove this commit if we want, if we expect to ever add the string back. However, I don't believe any string should lack context, and I don't see how to add context to a not-used string.

Some of those commits delete ressources that seems to be duplicate. If something is both a message and a menu entry (as in the "empty deck" example), I leave the duplication. However, when the only difference is that they are label of menu item in different windows, or that one if the title of an activity and the other one is the title of a window, I actually can't imagine reason to keep duplicate and give more works to our translators. Actually, it may even be more complex to explain to them why they would need to give distinct translation while the context can't explain it. Each deletion of duplicate is its own commit, so that if you disagree it can be removed. If you agree, on the other hand, we can move those commits in their own PR so that they can be considered separately.

I should note I realize that it's going to be a huge amount of work reviewing it if you expect to actually check every single comment I I added. I do not know which amount of review is needed here. On the one hand, this is NF, there should be absolutely no difference for user directly. On the other hand, if I makes an error, it may be hard to detect it later and it will lead to a lot of bad translation probably if the translators trust the context. And to be perfectly honest, over a thousand strings, it would be hard to imagine there is not a single error

### CODE REVIEW

File: AnkiDroid/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -231,7 +231,7 @@
             />
         <activity
             android:name="com.ichi2.anki.MyAccount"
-            android:label="@string/menu_my_account"
+            android:label="@string/button_sync"
```
Comment (david-allison-1) [inline AnkiDroid/src/main/AndroidManifest.xml:234]: ?

**Review by david-allison-1 — COMMENTED** [review]
This is going to conflict with many of the PRs that we have, I don't want to see this go to waste, but we need to figure out a way to get this in that's:

* Not relying on users manually formatting the XML to this style in the future
* Ideally not going to conflict heavily
* Ideally not going to affect git blame on these lines, as it makes history much harder to reason about (see: https://stackoverflow.com/a/57129540)
* Not reverting any changes in the master language files

The changes that this conflicts with are bringing new features into AnkiDroid, and for non-regular contributors we want this to be as simple as possible. When this goes in, any long-term forks will have significantly diverged from master.

@mikehardy - I'm unsure on this, but I feel https://github.com/ankidroid/Anki-Android/pull/7007/commits/47456d6e3ac744707442003b91bfc00fc43a9741 is causing more harm than good, as we lose any changes in the text when looking at the diff.

**Comment by mikehardy** [comment]
> @mikehardy - I'm unsure on this, but I feel [47456d6](https://github.com/ankidroid/Anki-Android/commit/47456d6e3ac744707442003b91bfc00fc43a9741) is causing more harm than good, as we lose any changes in the text when looking at the diff.

Really quickly - just on this one alone, I do not see a lot of value in it, I've let it sit because it makes me uncomfortable to just "no" things, but formatting for formatting's sake, when we don't have any automated formatting to make sure it never happens again? As with the previous formatting PR, I just don't see the point.

**Comment by Arthur-Milchior** [comment]
> > @mikehardy - I'm unsure on this, but I feel [47456d6](https://github.com/ankidroid/Anki-Android/commit/47456d6e3ac744707442003b91bfc00fc43a9741) is causing more harm than good, as we lose any changes in the text when looking at the diff.
> 
> Really quickly - just on this one alone, I do not see a lot of value in it, I've let it sit because it makes me uncomfortable to just "no" things, but formatting for formatting's sake, when we don't have any automated formatting to make sure it never happens again? As with the previous formatting PR, I just don't see the point.

First, please, really do not hesitate to say "no". I really don't want something to be accepted just because it is unconfortable to say no. If that help you, I want to let you know this commit required at most 5 minutes of work, probably far less, it's just the result of a emacs macro.

I would agree that this commit alone is totally useless if it's not considered as a step of this PR. The goal of the PR is to ensure that each non-english based user gets a good translation. To do this, translators need to get context to know what means the string they translate, so the comments need to be seen on crowdin. According to crowdin documentation, it means it should receive some xml with a comment field. I thought the easier way to do it is to insert comments in the XML files, even if there exists other way to do it (such as transforming the xml files before uploading them, by merging the ressource file with a comment file). 

At least two choices exists. Having the whole xml item and the ressource on the same line, or having them on different lines. I believe that it's more readable to have the ressource on its own line and the item on its own line. And even the comment on its own line. Furthermore, it leads to more readable commits; because this way, it's clear that the last commit modifies comments and does not touch the ressources. If the xml item and the ressources were on a single line, the ressource would be marked as modified while actually it was not.
Since I decided to put each logical element on a separate line, I preferred to make multiple commit, so that each commits makes one specific change. One commit add the new line. One other commit add the comment. Each commits gets easier to understand and to review. 
This is what lead to a commit which deals with reformatting; a commit which has no interest by itself, and whose only goal is to simplify the other commits of the same PR

**Comment by Arthur-Milchior** [comment]
> @mikehardy - I'm unsure on this, but I feel [47456d6](https://github.com/ankidroid/Anki-Android/commit/47456d6e3ac744707442003b91bfc00fc43a9741) is causing more harm than good, as we lose any changes in the text when looking at the diff.

Could you please give an example of "change in the text" that we are loosing ? I don't understand your point here.

File: AnkiDroid/src/main/AndroidManifest.xml (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/AndroidManifest.xml:234]: Thanks. That's strange. I'll proof read everything once all strings have a comment. Clearly not expected

**Comment by Arthur-Milchior** [comment]
> This is going to conflict with many of the PRs that we have

To be specific, ignoring my own PR, it conflicts with 6 PRs, including a 2018 PR with a lot of conflicts already ( 5134 ). 

I would expect those conflicts to be simple to solve, since you just need to take the strings you were adding in the other PRs and add them. Actually I'd argue it's nice that it'll conflict, this will force all current PR to have a comment before being merged. 

> I don't want to see this go to waste

If you have the time, I'd be interested in a related question. Is anyone else see the interest of this whole goal ? Believe that it'll ultimately improve the work of translator, and so the localizations of the app ? Because, I don't care about wasting something if it's useless. If it's not worth it, let's it go to waste, that's fine. I find it frightening to read that what you want is not to waste work, and not to read that what you want is to ensure better translation. That's the positive goal.

I should note that as long as we find any way to add commets, my work is not going to be wasted. I don't care about the syntax at all. If we find some other way to achieve this goal, of ensuring better translation, then that's perfect. In particular, with XML, I can easily create a script to extract the comment I wrote and rewrite the file in any syntax that we choose. 

So I want to emphasize, as long as translators can see the comments in crowdin, that's not wasted. (Even if ideally, I'd also like that comments are close to the english string, so this incentivises devs who add strings to directly as comment, because they see the comments nearby and imitate them)
 
> The changes that this conflicts with are bringing new features into AnkiDroid, and for non-regular contributors we want this to be as simple as possible. When this goes in, any long-term forks will have significantly diverged from master.

I should note that any admin can resolve conflict using github tools. If we merge this PR, I'd be willing to resolve the conflict myself if that's a matter of importance

**Comment by phu54321** [comment]
> Not relying on users manually formatting the XML to this style in the future

This needs an auto-formatter like [black](https://pypi.org/project/black/) on python. Since your requirements don't seem to be present on existing tools, maybe a new tool should be crafted. This might be an example: https://pastebin.pl/view/857ae20b

> Ideally not going to conflict heavily

You can register a pre-commit git hook to autoformat all string-related XML files every time before commit. Since every file is formatted the same way, there won't be much conflict d/t styling issues.

**Pull request closed without merging**